### PR TITLE
Luarocks pyupdater

### PR DESCRIPTION
**Pull request by teto**
###### Motivation for this change

I wanted to improve (mostly speed it up) the luarocks updater since I intend to run it more frequently. While this is not the case yet, I converted the shellscript to a python script reusing the kakoune/vim plugin infrastructure. Providing the user with an unifed interface across different ecosystems sound like a good idea where applicable.

I dont say my changes are elegant, I strived to minimize the changes in pluginupdate.py to get some feedback first. I believe a sounder OOP approach can be taken in subsequent steps.

cc @Flakebi 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by teto** [comment]
cc @timokau @Mic92 since you've worked the most on this part. I would like to rename the `Editor` class to something more generic too like `EcosystemUpdater`/`LanguageUpdater`.

File: maintainers/scripts/pluginupdate.py [hunk 1]
```diff
@@ -195,21 +203,46 @@ def __init__(
         name: str,
         root: Path,
         get_plugins: str,
+        # TODO modifier pour accepter un plugin
```
Comment (Mic92) [inline maintainers/scripts/pluginupdate.py:206]: TODO: traduire

Comment (teto) [reply maintainers/scripts/pluginupdate.py]: I removed the comment. I think generate_nix should be a member, as is done here. I've kept the argument to minimize changes.

**Comment by teto** [comment]
the global lua update had broken several packages, I've pinned versions to keep nixpkgs-review happy

**Comment by teto** [comment]
I've added more changes on a separate branch https://github.com/teto/nixpkgs/pull/4

**Comment by teto** [comment]
Kinda excited to use an unified interface to update different ecosystems. I will follow up with PRs if this gets merged but I dont want to merge it without the approval of any of the other persons using the update.py

**Comment by teto** [comment]
I've tested and vim update.py as well as luarocks updater work fine. I will merge and open another PR to improve the interface.

**Comment by teto** [comment]
wip follow up of https://github.com/NixOS/nixpkgs/pull/133296

**Comment by timokau** [comment]
I didn't / don't have the spare capacity to review this right now, but thank you for working on it. The lua neovim plugin ecosystem seems to be growing fast.

**Comment by teto** [comment]
I am working to improve luarocks2nix to better handle bare git repos so that the vim-plugin infra can handle lua plugin betters (for neovim). Right now lua plugins work but they have to be uploaded to luarocks.org and I dont expect vim plugin authors to do that, hence my changes.

**Comment by timokau** [comment]
I see. Thanks again for pushing this forward :)

**Pull request merged**